imageproxy.go

```diff
@@ -211,6 +225,41 @@ func (p *Proxy) allowed(r *Request) error {
 	return fmt.Errorf("request does not contain an allowed host or valid signature: %v", r)
 }
 
+// allowedContentType returns an allowed content type string to use in responses or "" if the
+// content type cannot be used.
+func (p *Proxy) allowedContentType(contentType string) string {
+	mediaType, _, _ := mime.ParseMediaType(contentType)
+	if mediaType == "" {
+		return ""
+	}
+
+	if len(p.ContentTypes) == 0 {
+		switch mediaType {
+		case "image/bmp", "image/cgm", "image/g3fax", "image/gif", "image/ief", "image/jp2",
```

Comment:
Having to list out all of these images types seems really messy and error prone.  You've already added support for `image/*` style wildcards below, so instead of treating an empty ContentTypes as special, just have the default flag value explicitly be `image/*`.  That results in safe default behavior (just images), but still allows for the existing behavior (all types), by simply passing an empty value for the contentTypes flag.

Or is there some other image/* subtype that is deemed unsafe that you're specifically trying to avoid?

Reply:
Since this is a security PR, I feel a lot better about using a more restrictive whitelist that's already been put to the test in other applications.

I can't say with much certainty that there are not dangerous image/* mimetypes. And I certainly can't say there will *never* be a dangerous image/* mimetype in the future (if there isn't already one).

So while there's nothing specifically that I'm aware of needing to avoid, I think it would be a security best practice to use a whitelist such as this one.

Reply:
where did this whitelist come from?

Reply:
Camo: https://github.com/atmos/camo/blob/master/mime-types.json